app/views/unattended/provisioning_templates/snippet/pkg_manager.erb

```diff
@@ -31,6 +31,10 @@ elif [ -f /etc/redhat-release ] ; then
   else
     PKG_MANAGER='yum'
   fi
+elif [ -f /etc/amazon-linux-release ]; then
+  PKG_MANAGER='dnf'
+elif [ -f /etc/system-release ]; then
```

Comment:
Out of interest: which other systems use `/etc/system-release`? And we already source `/etc/os-release` if it's present. Can't we rely on those variables?

Reply:
/etc/os-release is available, but the snippet doesn't  recognize yum in Amazon Linux 2. 

The current RHEL clones support /etc/system-release (alma,rocky,amzn). Fedora and RHEL obviously but those already have explicit support/detection.

I found the reference from this post on Redhat, [Proper way to extract Product, Release, Update, etc](https://access.redhat.com/discussions/2798501) which discusses OS detection

_3) System Release, vendor string: /etc/system-release (also via symlink, or legacy, /etc/fedora-release, /etc/redhat-release)_

The response links to a [sysconfcollect module](url) that identifies some 50+ distros near the bottom of the page. I didnt check any of them because I don't use them, though I don't expect support for them in Foreman/Katello.

Reply:
If we detect `/etc/system-release` can we use `dnf` instead of the `yum`?
Tested on Rocky & Alma and they both have `dnf`.